nj-sys/src/binding.rs

```diff
@@ -1,646 +1,258 @@
 /* automatically generated by rust-bindgen */
 
-pub const NAPI_VERSION: u32 = 5;
+pub const NAPI_VERSION: u32 = 7;
```

Comment:
I don't wanna preemptively add more scope to this task but I accidentally ended up pushing the wrong bindings earlier because I didn't run `make generate` in `js-sys`. If we're open to it, I'd love to move to using bindgen as a build dependency and generate these bindings at build time rather than having them in the source code. The pros for doing this are that when the next person to come along and update the N-API version, they'll pop in the new headers into the vendor directory, update the version and then just simply run `cargo build` in the parent directory and deal with whatever bugs exist. The cons of this are that build times will be longer and the jump-to-definition feature of [rust-analyzer may not work](https://github.com/rust-analyzer/rust-analyzer/issues/3767) for things in `nj-sys` :/.

Reply:
Initial version of the `node-bindgen` used `build.rs` to generate binding.  Main issue was build time which was long.  Hopefully, current rust compiler performance has improved that that is no longer concern.  

Also, there was bug in the `bindgen` the created inconsistent mapping between versions.  I think that could be mitigated by tied to specific version of `bindgen`

Reply:
> Initial version of the node-bindgen used build.rs to generate binding. Main issue was build time which was long. Hopefully, current rust compiler performance has improved that that is no longer concern.

Ah. Yeah. That's definitely a con. I'll play around with it and see what the compile time is like.